File: problem-solving/baekjoon/2670_연속부분최대곱.go
// https://www.acmicpc.net/problem/2670
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n int
	fmt.Fscanln(reader, &n)

	var realNumbers = make([]float64, n)
	for i := 0; i < n; i++ {
		fmt.Fscanln(reader, &realNumbers[i])
	}
	for i := 1; i < n; i++ {
		mul := realNumbers[i-1] * realNumbers[i]
		if realNumbers[i] < mul { // 이전의 값과 현재 값을 곱한 값이이 현재 값보다 더 크면 곱한 값으로 갱신
			realNumbers[i] = mul
		}
	}
	var answer float64
	for i := 0; i < n; i++ {
		if answer < realNumbers[i] {
			answer = realNumbers[i]
		}
	}
	fmt.Fprintf(writer, "%.3f\n", answer)
}

File: problem-solving/baekjoon/2670_연속부분최대곱.py
# https://www.acmicpc.net/problem/2670
import sys

if __name__ == "__main__":
    n = int(sys.stdin.readline())
    real_numbers = []
    for i in range(n):
        real_numbers.append(float(sys.stdin.readline()))
    for i in range(1, n):
        mul = real_numbers[i-1] * real_numbers[i]
        real_numbers[i] = max(mul, real_numbers[i])
    answer = max(real_numbers)
    print("{:.3f}".format(answer))